pre-commit-check.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+
+# This script is used to run checks before committing changes to the repository.
+# It is a good approximation of what CI will do.
+
+# Fail if any command fails
+set -e
+
+# Check formatting
+cargo +nightly fmt --all
+
+# Check clippy
+cargo clippy --all-features --all-targets -- -D warnings
+
+# Build all targets
+cargo build --all-targets
+
+# Run tests
+cargo test
+
+# Run wasm tests
+./crates/wasm-test-runner/run.sh
```

Comment:
not running those ignored expensive integration tests?

Reply:
I was a bit undecided here. I wanted a script to "quickly check" my local branch before pushing to GitHub and avoiding follow-up commits.

The main reason for extra follow-up commits is mostly **formatting** and **clippy**. So maybe we should limit this script to those.

Reply:
agreed, we can let testing be done in GH and only use this for clippy and formatting.